Task2: Done!

Feuerstarter opened the pull request:
https://feuerstarter.github.io/iTechArt-JS-Courses-2019/

Feuerstarter committed:
Task2: Done!

All the files to host website added;
updated readme

Feuerstarter committed:
fix: better work on mobile devices

viewport metatag added
fixed resolutions

anastasiatol reviewed (changes requested):
Некоторые элементы, которые выглядит как цельный элемент, поделены на части (header,  hr, который разделяет nav menu от main), из-за чего иногда виднеются белые щели (особенно в интернет эксплорере)

Чем меньше экран мобильного устройства, тем больше справа расстояние до контента, надпись на кнопках шире самой кнопки На мобильном хотелось бы на всю ширину устройства, можно нав меню скрыть за одной кнопкой, хедер неплохо было бы зафиксировать. Серч бывает обрезан снизу

Хотелось бы следующие элементы сделать более похожими на шаблон:
надпись NEW 
подложку под фото about me  
списки справа (которые aside)

Иконка не задана

При ширине экрана 1100 пикселей ширина содержимого слишком узкая , при уменьшении до 1099 резкий скачок. При размере 1100 можно фиксированную ширину установить, а не процент от размера от экрана

Шрифт в vh, все марджины и паддинги в процентах  => странное поведение при масштабировании, размер текста увеличить невозможно, все border-radiusы скругляются. При уменьшении высоты окна,  еще более странное поведение. Vh в основном для мобильных платформ

padding-left: 1%; padding-right: 1%; и др. такое лучше сокращать

По БЭМу, то что является блоком, не должен быть позиционирован, не задаются марджины
upper-border-radius-div-with-menu
search-form
lower-border-radius-div  
main-nav и др.
Присваивай второй класс как например search-form она и блок сама по себе, и может выступать элементом header. Поэтому просто добавляешь класс header__search-form и вот на это уже можно вешать эти свойства. 

Cелектор .section_new div  – не по БЭМу, нужен класс

header, main стили лучше не напрямую навешивать, а присвоить классы (вот как ты aside сделал например)

Feuerstarter committed:
Update: all the fixes required.

Fixes:
-better partial blocks display
-mobile version correct width
-"new", "about me" and "list" elements
-added icon
-width for PC
-font sizes, paddings & margins for low-res screens
-some code abbreviations
-better BEM syntax

Feuerstarter committed:
Updated README.md

Readme task2 screenshot updated.

anastasiatol reviewed (changes requested):
nav class="header__main-nav"
div class="main-nav__link

Ты везде опускал имя блока
nav class="header__main-nav main-nav"
div class="main-nav__link

И это ок, если на нем нету свойств. Но  часть этих свойств надо перенести на .main-nav{}, чтоб блок .main-nav сохранял свой внешний вид, когда будет использоваться без .header__main-nav

.header__main-nav {

grid-area: main-nav;
display: flex;
justify-content: left;
align-items: center;
margin-left: 7%;
}

.main {

grid-area: main;
}
блоку грид-эриа не задавать, это позиционирование

![default](https://user-images.githubusercontent.com/30156652/52177184-5982c780-27ce-11e9-8842-3ae0feb97212.PNG)
При разной ширине меняется вид картинки

Картинкам alt

Feuerstarter committed:
BEM fixes

-even better BEM syntax
-medium screen width adjusted
